celements-model/src/main/java/com/celements/model/context/DefaultModelContext.java

```diff
@@ -260,6 +265,22 @@ private String getDefaultLangFromConfigSrc(EntityReference ref) {
     }
   }
 
+  @Override
+  public XWikiDocument getOrCreateXWikiPreferenceDoc() {
+    DocumentReference docRef = new DocumentReference(getWikiRef().getName(), XWIKI_SPACE,
+        XWIKI_PREF_DOC_NAME);
+    return getModelAccess().getOrCreateDocument(docRef);
+  }
+
+  @Override
+  public XWikiDocument getOrCreateSpacePreferenceDoc() {
+    DocumentReference docRef = getCurrentDocRef().or(new DocumentReference(DEFAULT_DOC_NAME,
```

Comment:
1. use `EntityReferenceValueProvider` to get model defaults instead (this ensures support of xwiki.cfg configuration).
1. use `RefBuilder` to create new references instead. e. 
    `new RefBuilder().wiki(wikiName).space(spacename).doc(docname).build(ImmutableDocumentReference.class)`

Reply:
`DocumentReference.class` suffices